snippets/ObjectDestroy.hpp

```diff
@@ -1,64 +1,139 @@
-    {
-      VULKAN_HPP_ASSERT( m_owner && m_dispatch );
-      m_owner.destroy( t, m_allocationCallbacks, *m_dispatch );
-    }
-
-  private:
-    OwnerType                           m_owner               = {};
-    Optional<const AllocationCallbacks> m_allocationCallbacks = nullptr;
-    Dispatch const *                    m_dispatch            = nullptr;
-  };
-
-  class NoParent;
-
-  template <typename Dispatch>
-  class ObjectDestroy<NoParent, Dispatch>
+    VULKAN_HPP_ASSERT( m_owner && m_dispatch );
+    m_owner.destroy( t, m_allocationCallbacks, *m_dispatch );
+  }
+
+private:
+  OwnerType                           m_owner               = {};
+  Optional<const AllocationCallbacks> m_allocationCallbacks = nullptr;
+  Dispatch const *                    m_dispatch            = nullptr;
+};
+
+class NoParent;
+
+template <typename Dispatch>
+class ObjectDestroy<NoParent, Dispatch>
+{
+public:
+  ObjectDestroy() = default;
+
+  ObjectDestroy( Optional<const AllocationCallbacks> allocationCallbacks,
+                 Dispatch const & dispatch           VULKAN_HPP_DEFAULT_DISPATCHER_ASSIGNMENT ) VULKAN_HPP_NOEXCEPT
+    : m_allocationCallbacks( allocationCallbacks )
+    , m_dispatch( &dispatch )
   {
-  public:
-    ObjectDestroy() = default;
-
-    ObjectDestroy( Optional<const AllocationCallbacks> allocationCallbacks,
-                   Dispatch const & dispatch           VULKAN_HPP_DEFAULT_DISPATCHER_ASSIGNMENT ) VULKAN_HPP_NOEXCEPT
-      : m_allocationCallbacks( allocationCallbacks )
-      , m_dispatch( &dispatch )
-    {}
-
-    Optional<const AllocationCallbacks> getAllocator() const VULKAN_HPP_NOEXCEPT { return m_allocationCallbacks; }
-    Dispatch const & getDispatch() const VULKAN_HPP_NOEXCEPT { return *m_dispatch; }
-
-  protected:
-    template <typename T>
-    void destroy(T t) VULKAN_HPP_NOEXCEPT
-    {
-      VULKAN_HPP_ASSERT( m_dispatch );
-      t.destroy( m_allocationCallbacks, *m_dispatch );
-    }
-
-  private:
-    Optional<const AllocationCallbacks> m_allocationCallbacks = nullptr;
-    Dispatch const *                    m_dispatch            = nullptr;
-  };
+  }
+
+  Optional<const AllocationCallbacks> getAllocator() const VULKAN_HPP_NOEXCEPT
+  {
+    return m_allocationCallbacks;
+  }
+
+  Dispatch const & getDispatch() const VULKAN_HPP_NOEXCEPT
+  {
+    return *m_dispatch;
+  }
+
+protected:
+  template <typename T>
+  void destroy( T t ) VULKAN_HPP_NOEXCEPT
+  {
+    VULKAN_HPP_ASSERT( m_dispatch );
+    t.destroy( m_allocationCallbacks, *m_dispatch );
+  }
+
+private:
+  Optional<const AllocationCallbacks> m_allocationCallbacks = nullptr;
+  Dispatch const *                    m_dispatch            = nullptr;
+};
+
+//================================================================================================
+
+template <typename HandleType>
+class ObjectDestroyShared
+{
+public:
+  using handle_type = HandleType;
+  using parent      = parent_of_t<handle_type>;
```

Comment:
STL habit and Wisdom ancestry, since I was changing that, I needed to track all those types from parent of shared_handle, I will revise that

Reply:
after brief revision: those should stay, since ObjectDestroyShared and others are templated on handle and not on OwnerType. parent_type is shorter from `parent_of_t<handle_type>`, since I need to evaluate it once for each type I would like to keep it simple.

I renamed `parent` -> `parent_type` for consistency. `handle_type` is useful for type traits, since otherwise you'd need do the same monstrosity as in parent_of, since I couldn't get the owner type itself.
```
template <class HandleType>
struct parent_of
{
  using parent = typename std::conditional<has_parent<HandleType>, decltype( std::declval<default_vk_deleter<HandleType>>().getOwner() ), NoParent>::type;
};
```